aiida/manage/tests/pytest_fixtures.py

```diff
@@ -0,0 +1,147 @@
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+"""
+Collection of pytest fixtures using the TestManager for easy testing of AiiDA plugins.
+
+ * aiida_profile
+ * clear_database
+ * aiida_localhost
+ * aiida_code_factory
+
+"""
+from __future__ import division
+from __future__ import absolute_import
+from __future__ import print_function
+
+import tempfile
+import shutil
+import pytest
+from aiida.manage.tests import test_manager, get_test_backend
+
+
+@pytest.fixture(scope='session', autouse=True)
+def aiida_profile():
+    """Set up AiiDA test profile for the duration of the tests.
+
+    Note: scope='session' limits this fixture to run once per session. Thanks to ``autouse=True``, you don't actually
+     need to depend on it explicitly - it will activate as soon as you import it in your ``conftest.py``.
+    """
+    # create new TestManager instance
+    with test_manager(backend=get_test_backend()) as test_mgr:
+        yield test_mgr
+    # here, the TestManager instance has already been destroyed
+
+
+@pytest.fixture(scope='function', autouse=True)
+def clear_database(aiida_profile):  # pylint: disable=redefined-outer-name
+    """Clear the database after each test.
+
+    Note: Thanks to ``autouse=True``, this happens as soon as you import this fixure in your ``conftest.py``.
+    """
+    yield
+    # after the test function has completed, reset the database
+    aiida_profile.reset_db()
+
+
+@pytest.fixture(scope='function')
+def tempdir():
+    """Get a temporary directory.
+
+    E.g. to use as the working directory of an AiiDA computer.
+
+    :return: The path to the directory
+    :rtype: str
+
+    """
+    dirpath = tempfile.mkdtemp()
+    yield dirpath
+    # after the test function has completed, remove the directory again
+    shutil.rmtree(dirpath)
+
+
+@pytest.fixture(scope='function')
+def aiida_localhost(tempdir):  # pylint: disable=redefined-outer-name
+    """Get an AiiDA computer for localhost.
+
+    Usage::
+
+      def test_1(aiida_localhost):
+          name = aiida_localhost.get_name()
+          # proceed to set up code or use 'aiida_code_factory' instead
+
+
+    :return: The computer node
+    :rtype: :py:class:`aiida.orm.Computer`
+    """
+    from aiida.orm import Computer
+    from aiida.common.exceptions import NotExistent
+
+    name = 'localhost'
+
+    try:
+        computer = Computer.objects.get(name=name)
+    except NotExistent:
+        computer = Computer(
+            name=name,
+            description='localhost computer set up by test manager',
+            hostname=name,
+            workdir=tempdir,
+            transport_type='local',
+            scheduler_type='direct'
+        )
+        computer.store()
+        computer.configure()
+
+    return computer
+
+
+@pytest.fixture(scope='function')
+def aiida_code_factory(aiida_localhost):  # pylint: disable=redefined-outer-name
```

Comment:
I'm ambivalent on having only local computers: On the one hand, testing with remote computers probably isn't the greatest idea. But on the other hand, there might be some codes which can not be run locally.

Maybe to leave this option open, we should rename this to ``aiida_local_code_factory``?

Reply:
Will do!